packages/patterns/src/generatePatternStyles.ts

```diff
@@ -46,23 +59,26 @@ function truncate(val, pos) {
  * @param {Object} pattern.size Gradient size if not full size
  * @return {string} Minimal description for gradient.
  */
-function getGradientDescription({ type, rotation, center, size }) {
-  const sizeString = size
-    ? `ellipse ${truncate(100 * size.w, 2)}% ${truncate(100 * size.h, 2)}%`
-    : '';
-  const centerString = center
-    ? ` at ${truncate(100 * center.x, 2)}% ${truncate(100 * center.y, 2)}%`
-    : '';
-  switch (type) {
-    case 'radial':
+function getGradientDescription(pattern: Gradient) {
+  switch (pattern.type) {
+    case 'radial': {
+      const { size, center } = pattern as Radial;
```

Comment:
Wondering if we could define the gradient types in a way that the assertions here are not necessary, i.e. since `pattern.type === 'radial' we _know_ it's a `Radial`.

Reply:
Ah, just saw https://github.com/GoogleForCreators/web-stories-wp/pull/12200#discussion_r960552291

Reply:
Here it's actually because `Gradient` isn't a _supertype_ of `Linear` and `Radial`. Maybe I should change the types a bit so this will work. That probably requires interfaces though. I'll check to see if it can be improved.

Reply:
Fixed it in c986cafb77.